assertion/function/assertiontree/backprop_util.go

```diff
@@ -749,3 +750,22 @@ func CheckGuardOnFullTrigger(trigger annotation.FullTrigger) annotation.FullTrig
 	}
 	return trigger
 }
+
+// addAssignmentToConsumer updates the consumer with assignment entries for informative printing of errors
+func addAssignmentToConsumer(lhs, rhs ast.Expr, pass *analysis.Pass, consumer annotation.ConsumingAnnotationTrigger) (err error) {
+	var lhsExprStr, rhsExprStr string
+	if lhsExprStr, err = asthelper.PrintExpr(lhs, pass, true /* isShortenExpr */); err != nil {
+		return err
```

Comment:
Can we add more context to the error here? `fmt.Errorf("adding assignment to consumer: %w", err)`

Reply:
Good idea. Done!